src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,867 @@
+  fromℕ (aN *ₙ toℕ (b * c))   ≡⟨ cong fromℕ (sym (toℕ-homo-* a (b * c))) ⟩
+  fromℕ (toℕ (a * (b * c)))   ≡⟨ fromℕ∘toℕ (a * (b * c)) ⟩
+  (a * (b * c))               ∎
+  where
+  aN = toℕ a;   bN = toℕ b;   cN = toℕ c
+
+2*-*-assoc :  ∀ x y → (2* x) * y ≡ 2* (x * y)
+2*-*-assoc x y = begin
+  (2* x) * y     ≡⟨ cong (_* y) (2*≗2B* x) ⟩
+  (2B * x) * y   ≡⟨ *-assoc 2B x y ⟩
+  2B * (x * y)   ≡⟨ sym (2*≗2B* (x * y)) ⟩
+  2* (x * y)     ∎
+
+
+*-distribˡ-+ :  _DistributesOverˡ_ _*_ _+_
+*-distribˡ-+ a b c = begin
+  a * (b + c)                          ≡⟨ sym (fromℕ∘toℕ (a * (b + c))) ⟩
+  fromℕ (toℕ (a * (b + c)))            ≡⟨ cong fromℕ (toℕ-homo-* a (b + c)) ⟩
+  fromℕ (k *ₙ (toℕ (b + c)))           ≡⟨ cong (fromℕ ∘ (k *ₙ_))
+                                                         (toℕ-homo-+ b c) ⟩
+  fromℕ (k *ₙ (m +ₙ n))                ≡⟨ cong fromℕ (ℕp.*-distribˡ-+ k m n) ⟩
+  fromℕ (k *ₙ m +ₙ k *ₙ n)             ≡⟨ cong fromℕ $
+                                          cong₂ _+ₙ_ (sym (toℕ-homo-* a b))
+                                                     (sym (toℕ-homo-* a c))
+                                        ⟩
+  fromℕ (toℕ (a * b) +ₙ toℕ (a * c))   ≡⟨ cong fromℕ $
+                                            sym (toℕ-homo-+ (a * b) (a * c))
+                                        ⟩
+  fromℕ (toℕ (a * b + a * c))          ≡⟨ fromℕ∘toℕ (a * b + a * c) ⟩
+  a * b + a * c                        ∎
+  where
+  k = toℕ a;   m = toℕ b;   n = toℕ c
+
+
+*-distribʳ-+ :  _DistributesOverʳ_ _*_ _+_
+*-distribʳ-+ =  comm+distrˡ⇒distrʳ *-comm *-distribˡ-+
+
+*-distrib-+ :  _DistributesOver_ _*_ _+_
+*-distrib-+ =  (*-distribˡ-+ , *-distribʳ-+)
+
+2*-distrib : ∀ x y → 2* (x + y) ≡ 2* x + 2* y
+2*-distrib x y = begin
+  2* (x + y)            ≡⟨ 2*≗2B* (x + y) ⟩
+  2B * (x + y)          ≡⟨ *-distribˡ-+ 2B x y ⟩
+  (2B * x) + (2B * y)   ≡⟨ cong₂ _+_ eq eq' ⟩
+  2* x + 2* y           ∎
+  where
+  eq = sym (2*≗2B* x);   eq' = sym (2*≗2B* y)
+
+
+*-identityˡ : LeftIdentity 1B _*_
+*-identityˡ x = begin
+  1B * x                 ≡⟨ sym (fromℕ∘toℕ (1B * x)) ⟩
+  fromℕ (toℕ (1B * x))   ≡⟨ cong fromℕ (toℕ-homo-* 1B x) ⟩
+  fromℕ (1 *ₙ n)         ≡⟨ cong fromℕ (ℕp.*-identityˡ n) ⟩
+  fromℕ n                ≡⟨ fromℕ∘toℕ x ⟩
+  x                      ∎
+  where
+  n = toℕ x
+
+*-identityʳ : RightIdentity 1B _*_
+*-identityʳ x =  trans (*-comm x 1B) (*-identityˡ x)
+
+*-identity : Identity 1B _*_
+*-identity = (*-identityˡ , *-identityʳ)
+
+2B*x≡x+x :  ∀ x → 2B * x ≡ x + x
+2B*x≡x+x x = begin
+  2B * x            ≡⟨⟩
+  (1B + 1B) * x     ≡⟨ *-distribʳ-+ x 1B 1B ⟩
+  1B * x + 1B * x   ≡⟨ cong₂ _+_ (*-identityˡ x) (*-identityˡ x) ⟩
+  x + x             ∎
+
+2x≡x+x :  ∀ x → 2* x ≡ x + x
+2x≡x+x x =  trans (2*≗2B* x) (2B*x≡x+x x)
+
+[1+x]* :  ∀ x y → (1B + x) * y ≡ y + x * y
+[1+x]* x y = begin
+  (1B + x) * y     ≡⟨ *-distribʳ-+ y 1B x ⟩
+  1B * y + x * y   ≡⟨ cong (_+ (x * y)) (*-identityˡ y) ⟩
+  y + x * y        ∎
+
```

Comment:
This name (and the following ones) are non-standard.

Reply:
Do you mean the property names ``[1+x]*, *[1+x], suc*, *suc`` ?

For example, Matthew agreed with naming ``toℕ∘fromℕ`` instead of 
``toℕ∘fromℕ≗id``, and with some similar names.

Please, recommend it on behalf of the library team.
Really, need them to be ``[1+x]*y≡y+x*y``, etc.?